tests/unit_tests/deploy_solver_unit_tests.py

```diff
@@ -5,21 +5,82 @@
 import d3ploy.solver as solver
 
 
-def test_deploy_solver():
+def test_min_deploy_solver():
+    """ Tests if the minimize_number_of_deployment function works correctly
+        by deploying the smallest number of facilities while meeting diff"""
     for i in range(100):
         diff = -1.0 * random.uniform(0.01, 30.0)
         commod = {}
         for i in range(4):
             commod.update({str(i): random.uniform(0.1, 9.9)})
-        deploy_dict = solver.deploy_solver({'commod': commod}, 'commod', diff)
+        deploy_dict = solver.deploy_solver(commodity_dict={'commod': commod},
+                                           pref_dict={},
+                                           commod='commod',
+                                           diff=diff,
+                                           time=1)
         # actually deploy and see if it's good
         final_diff = diff
         for key, val in deploy_dict.items():
             final_diff += val * commod[key]
-
+        
         if final_diff > min(commod.values()):
             raise ValueError(
                 'The difference after deployment exceeds the capacity of the smallest deployable prototype')
-
         # if it didn't raise valueerror, we are good
         assert(True)
+
+def test_pref_solver_const():
+    """ Tests if the preference_deploy function works correctly
+        by deploying only the most preferred prototype to meet diff """
```

Comment:
I recommend adding in the docstring that this test is for the constant demand case (if I'm reading this properly).